### PR TITLE
Make nodes with parents not mergable

### LINKED ISSUE #2: Don't allow merging notes that already have parents.
If a node already has a parent, it should not be possible to combine it with other nodes, as the resulting structure would not be a tree.

### PR DESCRIPTION
**Pull request by dharasim**
Fixes #2

### CODE REVIEW

File: src/tree_annotation/database.cljs [hunk 1]
```diff
@@ -173,7 +173,16 @@
 
 (defn del-node [coord]
   (when (last-leaf-or-inner-node? coord)
-    (do (del-ancestors coord)
+    (do ; delete all ancestors of the node (parent, grand-parent, ...)
```
Comment (chfin) [inline src/tree_annotation/database.cljs]: no `do` necessary with `when`.

**Review by chfin — APPROVED** [review]
This fix works for now, but I think this should be solved by a different representation of the data that excludes these overlaps.

The correct data structure would be a list of trees, with every leaf being a singleton tree in the beginning. Merging would then remove a subsequence of trees from the list and insert a new tree in their place, where the old roots become the direct children of the new root.

File: src/tree_annotation/database.cljs (hunk 1 above)
Comment (dharasim) [reply src/tree_annotation/database.cljs]: Ahhh, I can never remember when I have to write do and when not. :) I'm sure there are a lot of other places in the code where there are unnecessary dos..

**Comment by dharasim** [comment]
> The correct data structure would be a list of trees, with every leaf being a singleton tree in the beginning. Merging would then remove a subsequence of trees from the list and insert a new tree in their place, where the old roots become the direct children of the new root.

Good suggestion!

**Pull request merged**